writers/mixedbatchwriter/mixedbatchwriter.go

```diff
@@ -199,39 +199,41 @@ func (m *batchManager[A, T]) flush(ctx context.Context) error {
 
 // special batch manager for insert messages that also keeps track of the total size of the batch
 type insertBatchManager struct {
-	batch             []*message.WriteInsert
-	writeFunc         func(ctx context.Context, messages message.WriteInserts) error
-	curBatchSizeBytes int64
-	maxBatchSizeBytes int64
-	logger            zerolog.Logger
+	batch     []*message.WriteInsert
+	writeFunc func(ctx context.Context, messages message.WriteInserts) error
+	bytes     writers.Capped[int64]
+	rows      writers.Capped[int64]
+	logger    zerolog.Logger
 }
 
 func (m *insertBatchManager) append(ctx context.Context, msg *message.WriteInsert) error {
-	if len(m.batch) == cap(m.batch) || m.curBatchSizeBytes+util.TotalRecordSize(msg.Record) > m.maxBatchSizeBytes {
-		if err := m.flush(ctx); err != nil {
-			return err
-		}
-	}
+	// we append data first not to call calculations twice
```

Comment:
Same issue with flushing after the limit was reached

Reply:
9dbac8d613ad90520e9c2daf08314d3528c00a72